Justfile

```diff
@@ -396,17 +401,76 @@ function terra_caseify()
       ;;
 
     terra_pyinstaller) # Deploy terra using pyinstaller
-      if ! Terra_Pipenv run sh -c "command -v pyinstaller" &> /dev/null; then
-        justify terra pipenv sync --dev
-      fi
-      Terra_Pipenv run pyinstaller --noconfirm "${TERRA_CWD}/freeze/terra.spec"
+      # if ! Terra_Pipenv run sh -c "command -v pyinstaller" &> /dev/null; then
+      #   justify terra pipenv sync --dev
+      # fi
+      local indirect
+      local app_prefix
+      local terra_apps
+      for app_prefix in TERRA_DSM; do #${TERRA_APP_PREFIXES[@]+"${TERRA_APP_PREFIXES[@]}"}; do
+        indirect="${app_prefix}_APPS[@]"
+        terra_apps=(${!indirect+"${!indirect}"})
+        array_to_python_ast_list_of_strings terra_apps ${terra_apps[@]+"${terra_apps[@]}"}
+        declare -x TERRA_APPS="${terra_apps}"
+
+        Terra_Pipenv run pyinstaller --noconfirm "${TERRA_CWD}/freeze/terra.spec"
+      done
       ;;
+    #   local app_prefix
+    #   local terra_rel
+    #   local indirect
+    #   local indirect2
+    #   local terra_apps
+
+    #   declare -x PYINSTALLER_PYTHON_VERSION=3.6.9
+    #   declare -x PYINSTALLER_VERSION=3.6
+    #   declare -x PYINSTALLER_IMAGE="vsiri/pyinstaller:${PYINSTALLER_PYTHON_VERSION}-${PYINSTALLER_VERSION}"
+
+    #   justify pyinstaller build
+
+    #   for app_prefix in ${TERRA_APP_PREFIXES[@]+"${TERRA_APP_PREFIXES[@]}"}; do
+    #     indirect="${app_prefix}_CWD"
+
+    #     local TERRA_PYINSTALLER_SRC_DIR="${TERRA_PYINSTALLER_SRC_DIR-${!indirect}}"
+    #     local TERRA_PYINSTALLER_DIST_DIR="${TERRA_PYINSTALLER_DIST_DIR-${TERRA_PYINSTALLER_SRC_DIR}/dist}"
+
+    #     indirect2="${app_prefix}_JUST_SETTINGS"
+    #     indirect2="${!indirect2-${JUST_SETTINGS}}"
+    #     terra_rel="$(relative_path "${!indirect}" "$(dirname "${indirect2}")")"
+    #     local VSI_COMMON_JUST_SETTINGS="${VSI_COMMON_JUST_SETTINGS-/src/${terra_rel}/$(basename "${indirect2}")}"
+
+
+    #     terra_rel="$(relative_path "${TERRA_CWD}" "${!indirect}")"
+
+    #     indirect="${app_prefix}_APPS[@]"
+    #     terra_apps=(${!indirect+"${!indirect}"})
+    #     array_to_python_ast_list_of_strings terra_apps ${terra_apps[@]+"${terra_apps[@]}"}
+    #     local DOCKER_COMPOSE_EXTRA_RUN_ARGS=(-e TERRA_APPS="${terra_apps}")
+
+    #     local terra_venv_volume="${COMPOSE_PROJECT_NAME}_terra-venv"
+    #     if ! docker inspect "${terra_venv_volume}" &> /dev/null; then
+    #       echo "Volume ${terra_venv_volume} does not exist. Needs to be initialized by running a terra command :TODO" >&2
+    #       JUST_IGNORE_EXIT_CODES=1
+    #       return 1
+    #     fi
+
+    #     indirect2="${app_prefix}_VENV_DIR"
+    #     local TERRA_PYINSTALLER_VENV_DIR="${!indirect2-}"
+    #     if [ -n "${TERRA_PYINSTALLER_VENV_DIR}" ]; then
+    #       TERRA_PYINSTALLER_VOLUMES=("${TERRA_PYINSTALLER_VENV_DIR}:/venv")
+    #     fi
+
+    #     justify pyinstaller run bash # pyinstaller /src/${terra_rel}/freeze/terra.spec
+    #   done
+    #   ;;
```

Comment:
Is there a reason to leave all the commented code above?

Reply:
Yes, I haven't fully abandoned getting the pyinstaller plugin working, I just can't get it working now. This was my latest attempt, but I had to swap the "old" manual way back in, because it at least "worked" (well enough)